source/standalone/tutorials/04_sensors/benchmark_cameras.py

```diff
@@ -0,0 +1,621 @@
+)
+
+parser.add_argument(
+    "--num_objects", type=int, default=10, required=False, help="How many objects to spawn into the scene."
+)
+
+parser.add_argument(
+    "--experiment_length",
+    type=int,
+    default=30,
+    required=False,
+    help="How many steps to average over",
+)
+
+AppLauncher.add_app_launcher_args(parser)
+args_cli = parser.parse_args()
+args_cli.enable_cameras = True
+
+if args_cli.visualize:
+    print("[WARNING]: You have selected to visualize. which means your benchmark results will not be meaningful.")
+    import matplotlib
+
+    matplotlib.use("Agg")  # Use a non-interactive backend
+
+if len(args_cli.ray_caster_visible_mesh_prim_paths) > 1:
+    print("[WARNING]: Ray Casting is only currently supported for a single, static object")
+# launch omniverse app
+app_launcher = AppLauncher(args_cli)
+simulation_app = app_launcher.app
+
+"""Rest everything follows."""
+
+import numpy as np
+import open3d as o3d
+import random
+import time
+import torch
+from matplotlib import pyplot as plt
+
+import omni.isaac.core.utils.prims as prim_utils
+
+import omni.isaac.lab.sim as sim_utils
+from omni.isaac.lab.assets import RigidObject, RigidObjectCfg
+from omni.isaac.lab.sensors.camera import Camera, CameraCfg, TiledCamera, TiledCameraCfg
+from omni.isaac.lab.sensors.ray_caster import RayCasterCamera, RayCasterCameraCfg, patterns
+from omni.isaac.lab.utils.math import convert_perspective_depth_image_to_orthogonal_depth_image, unproject_depth
+
+
+def create_camera_base(
+    camera_cls: type[Camera | TiledCamera],
+    camera_cfg: type[CameraCfg | TiledCameraCfg],
+    num_cams: int,
+    data_types: list[str],
+    height: int,
+    width: int,
+) -> Camera | TiledCamera | None:
+    """Generalized function to create a camera or tiled camera sensor."""
+    # Determine prim prefix based on the camera class
+    name = camera_cls.__name__
+
+    # Create the necessary prims
+    for idx in range(num_cams):
+        prim_utils.create_prim(f"/World/{name}_{idx:02d}", "Xform")
+
+    # If valid camera settings are provided, create the camera
+    if num_cams > 0 and len(data_types) > 0 and height > 0 and width > 0:
+        cfg = camera_cfg(
+            prim_path=f"/World/{name}_.*/{name}",
+            update_period=0,
+            height=height,
+            width=width,
+            data_types=data_types,
+            spawn=sim_utils.PinholeCameraCfg(
+                focal_length=24, focus_distance=400.0, horizontal_aperture=20.955, clipping_range=(0.1, 1e4)
+            ),
+        )
+        return camera_cls(cfg=cfg)
+    else:
+        return None
+
+
+def create_tiled_cameras(
+    num_cams: int = 2, data_types: list[str] = ["rgb", "depth"], height: int = 100, width: int = 120
+) -> TiledCamera | None:
+    """Defines the tiled camera sensor to add to the scene."""
+    return create_camera_base(TiledCamera, TiledCameraCfg, num_cams, data_types, height, width)
```

Comment:
NIT: I prefer named arguments when they get long (e.g. 2+ args) otherwise it's easy to mix them up